scipy/linalg/flapack_sym_herm.pyf.src

```diff
@@ -302,6 +302,46 @@ end subroutine <prefix2c>heevd_lwork
    end subroutine <prefix>sytrf_lwork
 
 
+   subroutine <prefix>sytrs(n,nrhs,a,lda,ipiv,b,ldb,info,lower)
+   
+   ! Solve A * X = B for symmetric A matrix after calling ?sytrf
+   
+     callstatement (*f2py_func)((lower?"L":"U"),&n,&nrhs,a,&lda,ipiv,b,&ldb,&info)
+     callprotoargument char*,F_INT*,F_INT*,<ctype>*,F_INT*,F_INT*,<ctype>*,F_INT*,F_INT*
+     
+     <ftype> intent(in), dimension(lda, n), check((lda >= n) && (n >= 0)) :: a
+     <ftype> intent(in,out,copy,out=x),dimension(ldb, nrhs) :: b
+     integer optional,intent(in),check(lower==0||lower==1) :: lower = 0
```

Comment:
I thought when you write `lower = 0` that `0` is the default value. Does it also need to be marked as `optional`?

Reply:
In the start of the statement there is already `optional`

Reply:
I know. But do you know what happens if you were to write `lower = 0` but _not_ include `optional`? Coming from a Python background, it confuses me that `optional` exists because:

- if you were to include only `optional` but not the default value, what would the default value be?
- what is the good of a default value if the argument is _not_ optional?

If both are required, and that's just the syntax, I accept it; I'm just curious.